weatherman.py

```diff
@@ -0,0 +1,194 @@
+                month_statistics['average_lowest_temperature'] += float(minimum_temperature)
+        except ValueError:
+            continue
+
+    return month_statistics
+
+
+def print_year_statistics(highest_temperature, highest_temperature_day, lowest_temperature, lowest_temperature_day, most_humidity, most_humidity_day):
+    print(f"Highest Temperature: {highest_temperature}°C on {highest_temperature_day}")
+    print(f"Lowest Temperature: {lowest_temperature}°C on {lowest_temperature_day}")
+    print(f"Most humidity {most_humidity}% on  {most_humidity_day} ")
+
+def print_month_statistics(mean_iteration, mean_humidity, average_lowest_temperature, average_highest_temperature):
+    if mean_iteration > 0:
+        average_mean_iteration = mean_humidity / mean_iteration
+        average_highest_temperature /= mean_iteration
+        average_lowest_temperature /= mean_iteration
+
+        print(f"Average Lowest Temperature: {average_lowest_temperature:.3f}")
+        print(f"Average Highest Temperature: {average_highest_temperature:.3f}")
+        print(f"Average Mean Humidity: {average_mean_iteration:.3f}%")
+    else:
+        print("No valid data found for the given month.")
+
+
+def generate_weather_bar_chart(weather_details):
+    for row in weather_details:
+        date = row['PKT']
+
+        try:
+            maximum_temperature = int(row['Max TemperatureC'])
+            minimum_temperature = int(row['Min TemperatureC'])
+            reset_code = Style.RESET_ALL
+
+            color_code = Fore.RED
+            bar = "+" * maximum_temperature
+            print(f"{color_code}{date} {bar:<25} {maximum_temperature}C{reset_code}")
+
+            color_code = Fore.BLUE
+            bar = "+" * minimum_temperature
+            print(f"{color_code}{date} {bar:<25} {minimum_temperature}C{reset_code}")
+
+        except ValueError:
+            continue 
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="Process weatherman data files.")
+    parser.add_argument(
+        "-e", "--weather_year",
+        help="For a given year, it will display the highest temperature and day,"
+             " lowest temperature and day, most humid day and humidity."
+    )
+    parser.add_argument(
+        "-a", "--weather_year_month",
+        help="For a given month, it will display the average highest temperature,"
+             " average lowest temperature, average mean humidity."
+    )
+    parser.add_argument(
+        "-c", "--weather_bar_chart",
+        help="For a given month, it will draw two horizontal bar charts on the console"
+             " for the highest and lowest temperature on each day. Highest in"
+             " red and lowest in blue."
+    )
+    return parser.parse_args()
+
+def main():
+    args = parse_arguments()
+    weather_year = None
+    weather_month = None
+
+    if args.weather_year_month:
+        weather_year, weather_month_input = args.weather_year_month.split('/')
+        try:
```

Comment:
Please handle the exception with their actual location

Reply:
is has to be splited before calling the function because first we have to convert number to months